metadata-integration/java/datahub-client/src/test/java/datahub/client/file/FileEmitterTest.java

```diff
@@ -0,0 +1,154 @@
+package datahub.client.file;
+
+import static org.junit.Assert.fail;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import com.fasterxml.jackson.core.exc.StreamReadException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DatabindException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.linkedin.data.DataMap;
+import com.linkedin.data.template.JacksonDataTemplateCodec;
+import com.linkedin.dataset.DatasetProperties;
+import com.linkedin.mxe.MetadataChangeProposal;
+
+import datahub.client.Callback;
+import datahub.client.MetadataWriteResponse;
+import datahub.event.MetadataChangeProposalWrapper;
+
+public class FileEmitterTest {
+  private final ObjectMapper objectMapper = new ObjectMapper();
+  private final JacksonDataTemplateCodec dataTemplateCodec = new JacksonDataTemplateCodec(objectMapper.getFactory());
+  private static final String RESOURCE_DIR = "src/test/resources";
+  private static final String GOLDEN_FILE_DIR = RESOURCE_DIR + "/golden_files";
+
+  @Rule
+  public TemporaryFolder tempFolder = new TemporaryFolder();
+
+  @Test
+  public void testFileEmitter() throws URISyntaxException, IOException {
+
+    String goldenJson = GOLDEN_FILE_DIR + "/mysql_mces_golden.json";
+    String tempRoot = tempFolder.getRoot().toString();
+    String outputFile = tempRoot + "/test.json";
+    FileEmitter emitter = new FileEmitter(FileEmitterConfig.builder().fileName(outputFile).build());
+    for (MetadataChangeProposal mcp : this.getMCPs(goldenJson)) {
+      emitter.emit(mcp);
+    }
+    emitter.close();
+
+    this.assertEqualJsonFile(goldenJson, outputFile);
+
+  }
+
+  private void assertEqualJsonFile(String file1, String file2) throws StreamReadException, DatabindException,
+      IOException {
+    TypeReference<Map<String, Object>[]> typeRef = new TypeReference<Map<String, Object>[]>() {
```

Comment:
Can we read the files as a List<Map<String,Object>> and do a .equals() on the list? If it works, it'll be simpler...

Reply:
Done